packages/cli/src/databases/migrations/common/1738709609940-CreateFolderTable.ts

```diff
@@ -0,0 +1,60 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+export class CreateFolderTable1738709609940 implements ReversibleMigration {
+	async up({ runQuery, escape, schemaBuilder: { createTable, column } }: MigrationContext) {
+		const workflowTable = escape.tableName('workflow_entity');
+		const workflowFolderId = escape.columnName('folderId');
+		const folderTable = escape.tableName('folder');
+		const folderId = escape.columnName('id');
+
+		await createTable('folder')
+			.withColumns(
+				column('id').varchar(36).primary.notNull,
+				column('name').varchar(128).notNull,
+				column('parentId').varchar(36).default(null),
+				column('projectId').varchar(36).notNull,
+			)
+			.withForeignKey('projectId', {
+				tableName: 'project',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			})
+			.withForeignKey('parentId', {
+				tableName: 'folder',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			})
+			.withIndexOn(['projectId', 'id'], true).withTimestamps;
+
+		await createTable('folder_tag')
```

Comment:
Should we create this now if we are not adding it in mvp? or are we adding it because it's a quick followup?

Just worried we add this now, but don't get to it any time soon.

Reply:
I would rather do a single migration here, because we can link all entities, which means the backend can start filtering by tags even though the FE is not ready. Actually in my follow up PR today I'm already using the tags to filter the folders by it.

Reply:
are we already 100% committed to folder tags though? if yes, then ignore me. but if no, I'll have to agree here that we should remove this from the migration until we have a very clear picture of what folder tags would look like.

Reply:
As far as I know we are 100% committed to it. It's on the hand off, we have the designs for it, etc. We just delayed it for a second phase because we have to meet a deadline. Thoughts @mutdmour

Reply:
AFAIK we are definitely committed and should be in the next phase following this, which we are planning to be a quick followup to the mvp we are building now.

Reply:
So I would agree to keep it